tests/test_pythonpackage.py

```diff
@@ -0,0 +1,133 @@
+
+import os
+import shutil
+import sys
+import subprocess
+import tempfile
+
+from pythonforandroid.pythonpackage import (
+    get_package_dependencies,
+    get_dep_names_of_package,
+    get_package_name,
+    _get_system_python_executable,
+)
+
+
+def test_get_package_name():
+    assert get_package_name("PiLLOw") == "Pillow"
+
+
+def test_get_package_dependencies():
+    # Check that jinja2 is returned as direct dep:
+    assert(len([dep for dep in get_package_dependencies("python-for-android")
+                if "jinja2" in dep]) > 0)
+    # Check that MarkupSafe is returned as indirect dep of jinja2:
+    assert [
+        dep for dep in get_package_dependencies(
+            "python-for-android", recursive=True
+        )
+        if "MarkupSafe" in dep
+    ]
+    # Check that colorama is returned without the install condition when
+    # just getting the names:
+    assert "colorama" in get_dep_names_of_package("python-for-android")
+
+
+def test__get_system_python_executable():
+    # This is kind of absurd that we need this to run the test at all,
+    # but we can't test this inside tox's virtualenv:
+    pybin = _get_system_python_executable()
+
+    # The python binary needs to match our major version to be correct:
+    pyversion = subprocess.check_output([
+        pybin, "-c", "import sys; print(sys.version)"
+    ], stderr=subprocess.STDOUT)
+    try:
+        pyversion = pyversion.decode("utf-8", "replace")
+    except AttributeError:
+        pass
+    assert(pyversion.strip() == sys.version.strip())
+
+    # Helper function to run our function:
+    def run__get_system_python_executable(pybin):
+        # We want to see what _get_system_python_executable() does given
+        # a specific python, so we need to make it import it and run it:
+        cmd = [
+            pybin,
+            "-c",
+            "import importlib\n"
+            "import json\n"
+            "import os\n"
+            "import sys\n"
+            "sys.path = [os.path.dirname(sys.argv[1])] + sys.path\n"
+            "m = importlib.import_module(\n"
+            "    os.path.basename(sys.argv[1]).partition('.')[0]\n"
+            ")\n"
+            "print(m._get_system_python_executable())",
+            os.path.join(os.path.dirname(__file__), "..",
+                         "pythonforandroid", "pythonpackage.py"),
+        ]
+        # Actual call to python:
+        try:
+            result = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError("call failed, with output: " + str(e.output))
+        try:
+            return result.decode("utf-8", "replace").strip()
+        except AttributeError:
+            return result.strip()
+
+    # Check that in system-wide python, the system-wide python is returned:
+    try:
+        p1 = os.path.normpath(run__get_system_python_executable(pybin))
+        p2 = os.path.normpath(pybin)
+        assert(p1 == p2)
```

Comment:
assert parenthesis is misleading, it's not a function but a keyword

Reply:
haha, sorry about that. i do these all the time because subconsciously I think it's a wrong decision/ugly on python's part (should have been turned to a function with python 3 as `print` was) but I know about the confusing behavior if a message is appended. I'll remove it :+1:

Reply:
Yes I also think it's a sad feature of Python3. I'm not sure why they fixed the `print()` and not `assert`. I hope there're good reasons at least.
Thanks for addressing anyay